### ISSUE TITLE
How to set timeout options for AWS Service calls per Service

### ISSUE THREAD

**Issue by ronak2121** [aws/aws-sdk-js-v3#2028]
### Is your feature request related to a problem? Please describe.
AWS JS SDK v2 had an easy way to set the timeout value for all AWS service calls (as part of httpOptions). This seems to be missing in AWS JS SDK v3 (or at least I can't find it through all of your documentation).

### Describe the solution you'd like
A way to define timeout values per AWS Service so we can control latency for calls from our Lambda@Edge Lambdas.

### Describe alternatives you've considered
None

### Additional context
None

**Comment by ronak2121** [comment]
I think the answer is socketTimeout by using NodeHTTPHandlerOptions right?

**Comment by trivikr** [comment]
> I think the answer is socketTimeout by using NodeHTTPHandlerOptions right?

**TL;DR** answer: yes.

The documentation of [httpOptions.timeout](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#constructor-property) from AWS SDK for JavaScript (v2)

> **timeout [Integer] —** Sets the socket to timeout after timeout milliseconds of inactivity on the socket. Defaults to two minutes (120000).

The v2 sets this value as a timeout of http.request in https://github.com/aws/aws-sdk-js/blob/10931414d507641df505eeb05c6f2947a51cad46/lib/http/node.js#L82-L88

Node.js API docs: https://nodejs.org/api/http.html#http_request_settimeout_timeout_callback

The equivalent option in AWS SDK for JavaScript (v3) is `socketTimeout` in HttpOptions.
https://github.com/aws/aws-sdk-js-v3/blob/5ec9314eaa21c45f054b66289004e63dfceb7864/packages/node-http-handler/src/set-socket-timeout.ts#L3-L9

It's documented in https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/nodehttphandleroptions.html#sockettimeout

**Comment by ronak2121** [comment]
Sounds good. Closing this.
